### PR TITLE
Form Validation

### PR DESCRIPTION
**Pull request by Besufekad-HAZ**
- updated the form validation
- fix reg expression and upper case check
- checked stylint and eslint errors and fixed it.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -25,3 +25,34 @@ ham_menu2.addEventListener("click", () => {
...
+//     //  else if (name.length == 0) {
+//     //    event.preventDefault();
+//     //    msg_info.innerHTML = "Please write your name";
+//     else {
+//        form.submit();
+//     }
+//   });
+// }
+
+// validation();
```
Comment (tresorsawasawa) [inline index.js]: - [ ] KIndly make sure you remove all the commented and unused code lines from your code base for **[best practice reasons](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md)**. This makes your project look messy.

File: style.css [hunk 2]
```diff
@@ -426,6 +428,168 @@ textarea:focus {
...
+
+.incorrect {
+  padding-left: 10px;
+  color: rgb(255, 0, 0);
+  font-size: 15px;
+  font-weight: 200;
+  position: relative;
+  bottom: 22px;
+} */
+
```
Comment (tresorsawasawa) [inline style.css]: - [ ] As I mentioned in the previous comment, KIndly make sure you remove all the commented and unused code lines from your code base for **[best practice reasons](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md)**. This makes your project look messy.

File: index.html [hunk 3]
```diff
@@ -277,27 +277,30 @@ <h3 class="my_name">Get Started</h2>
...
               <ul class="list_forms">
                 <div class="name_email">
                   <li class="form_input">
                     <label>
                       Your name<input placeholder="Your Name" type="text" id="name"
-                        name="name" maxlength="30" required>
+                        name="name" maxlength="30" required pattern="([A-ZΆ-ΫÀ-ÖØ-Þ][A-ZΆ-ΫÀ-ÖØ-Þa-zά-ώß-öø-ÿ]{1,19} ?){1,10}" onchange="storeForm()">
                     </label>
                   </li>
+                  <!-- <p id="reply_name"></p>  -->
```
Comment (tresorsawasawa) [inline index.html]: - [ ] KIndly make sure you remove all the commented and unused code lines from your code base for **[best practice reasons](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md)**. This makes your project look messy.

File: index.html [hunk 4]
```diff
@@ -277,27 +277,30 @@ <h3 class="my_name">Get Started</h2>
...
                     <label>
                       Your email
                       <input
-                        placeholder="Your Email"
-                        id="mail" type="email" name="email" required>
+                        placeholder="Your Email" class="email_bg"
+                        id="mail" type="email" placeholder="email@domain.com" name="email" required>
                     </label>
                   </li>
+                   <!-- <p id="reply_mail"></p>  -->
```
Comment (tresorsawasawa) [inline index.html]: - [ ] KIndly make sure you remove all the commented and unused code lines from your code base for **[best practice reasons](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md)**. This makes your project look messy.

File: index.html [hunk 5]
```diff
@@ -218,6 +218,7 @@ <h4 class="proj_name">Project name goes here</h4>
               </div>
             </div>
           </div>
+          <!-- <div class="modal-overlay"></div> -->
```
Comment (tresorsawasawa) [inline index.html:221]: - [ ] As mentioned in the above comments, KIndly make sure you remove all the commented and unused code lines from your code base for **[best practice reasons](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md)**. This makes your project look messy.
You can check in all your files and apply this approach to them. I know you can do that :muscle: :muscle:

**Review by tresorsawasawa — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES :recycle::recycle: 

Hi @Besufekad-HAZ  and @ShiroYaksha90 
Good job so far! I know you worked hard to implement this project, and I really appreciate the effort you put in! But there are some issues that you still need to work on to go to the next project but you are almost there! 

almost-there

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Your design looks good :100: :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- Follow GitHub flow :heavy_check_mark: 

## Required changes :recycle: :recycle: 
- [ ] Check the comments under the review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear. 
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Deepakdanger — APPROVED** [review]
Hi @Besufekad-HAZ , 

### Status: Approved ✔️

Your project is complete!  There is nothing else to say other than... it's time to merge it  🚢 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**